restli-client/src/main/java/com/linkedin/restli/client/uribuilders/AbstractRestliRequestUriBuilder.java

```diff
@@ -105,11 +105,7 @@ protected void appendQueryParams(UriBuilder b)
 
   protected final void appendAssocKeys(UriBuilder uriBuilder)
   {
-    if (_assocKey == null)
-    {
-      throw new IllegalArgumentException("_assocKey is null");
-    }
-    if (_assocKey.getNumParts() != 0)
+    if (_assocKey != null && _assocKey.getNumParts() != 0)
```

Comment:
We are not removing the null check. We are just not logging an error for assocKey being null. This is the expected behavior as assocKey will be null for non association resources. Requestbuilders are creating a dummy CompoundKey to avoid this error, which is wasteful.